common/src/main/java/com/hedera/block/common/utils/Preconditions.java

```diff
@@ -16,25 +16,36 @@
 
 package com.hedera.block.common.utils;
 
-import java.util.Objects;
+import edu.umd.cs.findbugs.annotations.NonNull;
 
 /** A utility class used to assert various preconditions. */
-public final class Preconditions { // @todo(381) change the APIs to accept non-null error messages
+public final class Preconditions {
+    private static final String DEFAULT_NOT_BLANK_MESSAGE = "The input String is required to be non-blank.";
+    private static final String DEFAULT_REQUIRE_POSITIVE_MESSAGE = "The input number [%d] is required to be positive.";
+    private static final String DEFAULT_GT_OR_EQ_MESSAGE =
+            "The input number [%d] is required to be greater or equal than [%d].";
+    private static final String DEFAULT_REQUIRE_IN_RANGE_MESSAGE =
+            "The input number [%d] is required to be in the range [%d, %d] boundaries included.";
+    private static final String DEFAULT_REQUIRE_WHOLE_MESSAGE =
+            "The input number [%d] is required to be a whole number.";
+    private static final String DEFAULT_REQUIRE_POWER_OF_TWO_MESSAGE =
+            "The input number [%d] is required to be a power of two.";
+
     /**
      * This method asserts a given {@link String} is not blank, meaning it is
      * not {@code null} or does not contain only whitespaces as defined by
      * {@link String#isBlank()}. If the given {@link String} is not blank, then
      * we return it, else we throw {@link IllegalArgumentException}.
```

Comment:
Another slight nitpick.  Javadoc treats the first "sentence" specially (defined as everything up to a `.` followed by a newline), and that should be a short "summary" line for a method.
```suggestion
     * This method asserts a given {@link String} is not blank.
     * A blank string is {@code null} or contains only whitespaces as defined by
     * {@link String#isBlank()}.<br/>If the given {@link String} is not blank,
     * then we return it, otherwise we throw {@link IllegalArgumentException}.
```

Reply:
✅ Thanks for that, I've made this improvement.